src/components/common/UserProfile/UserColonies.tsx

```diff
@@ -20,23 +21,39 @@ const displayName = 'common.UserProfile.UserColonies';
 const MSG = defineMessages({
   currentUserNoColonies: {
     id: `${displayName}.currentUserNoColonies`,
-    defaultMessage: `It looks like you have not joined any Colonies yet. You’ll need an invite link to join one. Ask your community for a link or {createColonyLink}.`,
+    defaultMessage: `It looks like you have not joined any Colonies yet. You'll need an invite link to join one. Ask your community for a link or {createColonyLink}.`,
   },
   otherUserNoColonies: {
     id: `${displayName}.otherUserNoColonies`,
-    defaultMessage: `It looks like {friendlyUsername} hasn't joined any Colonies yet. You’ll might want to send them an invite link from a Colony you're part of.`,
+    defaultMessage: `It looks like {friendlyUsername} hasn't joined any Colonies yet. You might want to send them an invite link from a Colony you're part of.`,
   },
   createColonyLink: {
     id: `${displayName}.createColonyLink`,
     defaultMessage: `create a new colony`,
   },
+  loadingColonies: {
+    id: `${displayName}.loadingColonies`,
+    defaultMessage: 'Loading colonies...',
+  },
 });
 
 const UserColonies = ({
   user: { walletAddress, watchlist, name, profile },
 }: Props) => {
-  const { user: currentUser } = useAppContext();
+  const { user: currentUser, walletConnecting } = useAppContext();
   const isCurrentUser = currentUser?.walletAddress === walletAddress;
+
+  if (walletConnecting) {
+    return (
+      <div className={styles.loader}>
+        <SpinnerLoader
```

Comment:
When the wallet is connecting, this will default to MSG.otherUserNoColonies, and so we get this flashing of text in the event the other message should be shown once the wallet connects. I think it's preferable to show a loading state while the wallet is connecting, hence the inclusion of a spinner here and moving the `walletConnecting` state into appContext.

Reply:
I actually appreciate you explaining the more "funky" parts of the implementation. 

I personally leave actual comments inside the file, but I think your way of doing it might be preferable overall as it enables discussion. _(ie: comments chain)_

I want to encourage this going forward as a git / PR style @jakubcolony @ArmandoGraterol @Julianahlk